test/e2e_node/cpu_manager_test.go

```diff
@@ -751,12 +856,251 @@ func isSMTAlignmentError(pod *v1.Pod) bool {
+			var allocatableCPU int64
+
+			ginkgo.BeforeEach(func(ctx context.Context) {
+				// track all pods created by a It() block
+				podMap = make(map[string]*v1.Pod)
+
+				_, allocatableCPU, _ = getLocalNodeCPUDetails(ctx, f)
+				if allocatableCPU < 4 {
+					e2eskipper.Skipf("Skipping CPU Manager tests since the CPU allocatable < 4")
+				}
+			})
+
+			ginkgo.AfterEach(func(ctx context.Context) {
+				deletePodsAsync(ctx, f, podMap)
+			})
+
+			ginkgo.It("should allocate exclusive CPUs for each container", func(ctx context.Context) {
+				reservedCPUs := cpuset.New(0)
+
+				updateKubeletConfigIfNeeded(ctx, f, configureCPUManagerInKubelet(oldCfg, &cpuManagerKubeletArguments{
+					policyName:         string(cpumanager.PolicyStatic),
+					reservedSystemCPUs: reservedCPUs,
+				}))
+
+				cpuCount := 1
+				ctnAttrs := []ctnAttribute{
+					{
+						ctnName:    "gu-container-1cpu-1",
+						cpuRequest: fmt.Sprintf("%dm", 1000*cpuCount),
+						cpuLimit:   fmt.Sprintf("%dm", 1000*cpuCount),
+						command:    "/bin/sleep 1d", // 1d is functionally forever for this test
+					},
+					{
+						ctnName:    "gu-container-1cpu-2",
+						cpuRequest: fmt.Sprintf("%dm", 1000*cpuCount),
+						cpuLimit:   fmt.Sprintf("%dm", 1000*cpuCount),
+						command:    "/bin/sleep 1d", // 1d is functionally forever for this test
+					},
+				}
+				pod := makeCPUManagerPod("gu-pod-multicnt-1", ctnAttrs)
+				pod = e2epod.NewPodClient(f).CreateSync(ctx, pod)
+				podMap[pod.Name] = pod
+
+				ginkgo.By("checking if the expected cpuset was assigned")
+
+				gomega.Eventually(ctx, func(ctx context.Context) error {
+					for _, cnt := range pod.Spec.Containers {
+						ginkgo.By(fmt.Sprintf("validating the container %s on Gu pod %s", cnt.Name, pod.Name))
+
+						stdout, _, err := execCommandInContainer(f, pod.Name, cnt.Name, "/bin/cat", cgroupCpusetPath)
+						if err != nil {
+							return fmt.Errorf("failed to get command output from container [%s] of pod [%s]: %w", cnt.Name, pod.Name, err)
+						}
+
+						cpus, err := cpuset.Parse(strings.TrimSpace(stdout))
+						if err != nil {
+							return fmt.Errorf("parsing cpuset from logs for [%s] of pod [%s]: %w", cnt.Name, pod.Name, err)
+						}
+
+						if cpus.Size() != cpuCount {
+							return fmt.Errorf("expected cpu set size == %d, got %q", cpuCount, cpus.String())
+						}
+					}
+					return nil
+				}).WithPolling(5 * time.Second).WithTimeout(30 * time.Second).ShouldNot(gomega.HaveOccurred())
+
+				ginkgo.By("checking if the shared pool changed as expected")
+				// TODO: check the actual set composition in addition to the size?
+
+				gomega.Eventually(ctx, func(ctx context.Context) error {
+					node, ok, err := fetchLocalTestNode(ctx, f)
+					if err != nil {
+						return err
+					}
+					if !ok {
+						return fmt.Errorf("local node not ready yet")
+					}
+					allocatableCPUQty := node.Status.Allocatable[v1.ResourceCPU]
+					allocatableCPU := allocatableCPUQty.Value()
```

Comment:
We populated `allocatableCPU` using `getLocalNodeCPUDetails` with allocatable CPUs on the nodes in the BeforeEach ([here](https://github.com/kubernetes/kubernetes/pull/120754/commits/a23f3aec439005fe5a67a2e52061a3f67ae04a5e#diff-9f37461eb6db1067ff11d6118e4ea593c117772b66cb12d4725e52f9419384ecR980)). Why don't we reuse that rather than getting that again from node status?

Also, we have a case of variable shadowing here.